docs/sources/getting-started/logcli.md

```diff
@@ -430,3 +434,27 @@ Flags:
 Args:
   <matcher>  eg '{foo="bar",baz=~".*blip"}'
 ```
+
+### LogCLI `--stdin` usage
+
+You can consume log lines from your `stdin` instead of Loki servers.
+
+Say you have log files in your local, and just want to do run some LogQL queries for that, `--stdin` flag can help.
+
+**NOTE: Currently it doesn't support any type of metric queries**
+
+You may have to use `stdin` flag for several reasons
+1. Quick way to check and validate a LogQL expressions.
+2. Learn basics of LogQL with just Log files and `LogCLI`tool ( without needing set up Loki servers, Grafana etc.)
+3. Easy discussion on public forums. Like Q&A, Share the LogQL expressions.
+
+**NOTES on Usage**
+1. `--limits` flag doesn't have any meaning when using `--stdin` (use pager like `less` for that)
+1. Be aware there are no **labels** when using `--stdin`
+   - So stream selector in the query is optional e.g just `|="timeout"|logfmt|level="error"` is same as `{foo="bar"}|="timeout|logfmt|level="error"`
+
+**Examples**
+1. Line filter - `cat mylog.log | logcli --stdin query '|="too many open connections"'`
+2. Label matcher - `echo 'msg="timeout happened" level="warning"' | logcli --stdin query '|logfmt|level="warning"'`
+3. Different parsers (logfmt, json, pattern, regexp) - `cat mylog.log | logcli --stdin query '|pattern <ip> - - <_> "<method> <uri> <_>" <status> <size> <_> "<agent>" <_>'`
```

Comment:
Hm, this feels a little odd. Do you think we could also support `pattern <ip>` etc without the leading `|`?

Reply:
Hmm. Actually thought about that but then that may confuse other things like line filter. (e.g: how do differentiate |= and |~). We could remove leading `|` only for pattern (and some other things) but then it may also confuses people as now they have remember where should I use leading `|` and where shouldn't I. 

So to keep it simple with first iteration, we decided to go with just making label selector optional and everything remains same as far as LogQL query is concerned!

Reply:
Ah yes. You are right!